### PR TITLE
Implement node_info

### PR DESCRIPTION
**Pull request by dleutenegger**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -264,8 +267,26 @@ impl LightningNode {
...
+            node_pubkey: node_state.id,
+            peers: node_state.connected_peers,
+            channels_info: ChannelsInfo {
+                num_channels: 0,
+                num_usable_channels: 0,
+                local_balance: (node_state.channels_balance_msat + node_state.onchain_balance_msat)
+                    .to_amount_down(&rate),
+                inbound_capacity: node_state.max_receivable_msat.to_amount_down(&rate),
+                outbound_capacity: node_state.max_payable_msat.to_amount_down(&rate),
+                total_channel_capacities: node_state.max_payable_msat.to_amount_down(&rate),
```
Comment (danielgranhao) [inline src/lib.rs:287]: This value (`total_channel_capacities`) refers to the total channel capacities. I see no way of us getting this value from the SDK's interface, but I'm also not seeing why we would need it. I propose we remove it.

File: src/lib.rs [hunk 2]
```diff
@@ -264,8 +267,26 @@ impl LightningNode {
...
+        let rate = self.get_exchange_rate();
+
+        Ok(NodeInfo {
+            node_pubkey: node_state.id,
+            peers: node_state.connected_peers,
+            channels_info: ChannelsInfo {
+                num_channels: 0,
+                num_usable_channels: 0,
+                local_balance: (node_state.channels_balance_msat + node_state.onchain_balance_msat)
+                    .to_amount_down(&rate),
```
Comment (danielgranhao) [inline src/lib.rs:284]: I wouldn't put the sum of these 2 balances in here. The name `local_balance` could indeed refer to both balances, but it breaks our existing model/assumptions. We have to think about how to manage the onchain balance, but for now, I would just do the following:
```suggestion
                local_balance: node_state.channels_balance_msat.to_amount_down(&rate),
```
We could already expose the onchain balance as a separate field, leaving to the consumer the decision to display the sum or not.

File: src/lib.rs [hunk 3]
```diff
@@ -264,8 +267,26 @@ impl LightningNode {
...
+            msg: e.to_string(),
+        })?;
+        let rate = self.get_exchange_rate();
+
+        Ok(NodeInfo {
+            node_pubkey: node_state.id,
+            peers: node_state.connected_peers,
+            channels_info: ChannelsInfo {
+                num_channels: 0,
+                num_usable_channels: 0,
```
Comment (danielgranhao) [inline src/lib.rs:282]: If we keep this here, please add a comment on both lines mentioning the deprecation of the fields, as well as in the UDL.

File: src/lib.rs (hunk 1 above)
Comment (danielgranhao) [reply src/lib.rs:287]: Hum... I don't quite get what that field means. It seems to be a value about the LSP, not about our current existing channels

Comment (dleutenegger) [reply src/lib.rs:287]: I'm not sure if I understand this correctly, but wouldn't `total_channel_capacities` be the sum of `inbound_capacity` and `outbound_capacity`

Comment (danielgranhao) [reply src/lib.rs:287]: Besides channel reserves, there are other amounts that are not included in the `inbound`/`output` capacity, such as commit fees, so no, unfortunately we can't get the overall channel capacity that way

**Review by danielgranhao — APPROVED** [review]

**Pull request merged**